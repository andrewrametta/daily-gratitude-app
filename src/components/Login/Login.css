form input {
  margin-bottom: 15px;
  margin-top: 5px;
  margin-left: 10px;
  margin-right: 10px;
}
