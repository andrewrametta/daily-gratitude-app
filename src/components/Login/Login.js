import React, { Component } from "react";
import TokenService from "../../services/token-service";
import AuthAPIService from "../../services/auth-api-service";
import Context from "../../Context";
import "./Login.css";

class LoginForm extends Component {
  state = {
    error: null,
  };
  static contextType = Context;

  handleLogin = (e) => {
    e.preventDefault();
    const { username, password } = e.target;
    const user = { username: username.value, password: password.value };
    this.setState({ error: null });
    AuthAPIService.loginUser(user)
      .then((loginResponse) => {
        TokenService.saveAuthToken(loginResponse.authToken);
        AuthAPIService.getDay().then((days) => {
          this.context.setDays(days);
        });
        this.props.history.push("/dashboard");
      })
      .catch((res) => {
        this.setState({ error: res.error });
      });
  };

  render() {
    return (
      <div>
        <section>
          <h3>Login</h3>

          <form className="Login-form" onSubmit={this.handleLogin}>
            {this.state.error && <p className="error">{this.state.error}</p>}
            <div>
              <label className="label" htmlFor="username">
                Username{" "}
              </label>
              <input
                type="text"
                name="username"
                id="username"
                defaultValue="DemoUser"
              />
            </div>
            <div>
              <label className="label" htmlFor="password">
                Password{" "}
              </label>
              <input
                type="password"
                name="password"
                id="password"
                defaultValue="P@ssword1234"
              />
            </div>
            <button className="button-login" type="submit">
              Login
            </button>
          </form>
        </section>
      </div>
    );
  }
}

export default LoginForm;
