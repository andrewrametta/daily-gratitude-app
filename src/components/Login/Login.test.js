import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Login from "./Login";
// setup file
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe(`Login component`, () => {
  it("renders the complete page", () => {
    const wrapper = shallow(<Login />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
