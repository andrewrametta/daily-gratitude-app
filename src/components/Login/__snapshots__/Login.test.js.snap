// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders the complete page 1`] = `
<div>
  <section>
    <h3>
      Login
    </h3>
    <form
      className="Login-form"
      onSubmit={[Function]}
    >
      <div>
        <label
          className="label"
          htmlFor="username"
        >
          Username
           
        </label>
        <input
          defaultValue="DemoUser"
          id="username"
          name="username"
          type="text"
        />
      </div>
      <div>
        <label
          className="label"
          htmlFor="password"
        >
          Password
           
        </label>
        <input
          defaultValue="P@ssword1234"
          id="password"
          name="password"
          type="password"
        />
      </div>
      <button
        type="submit"
      >
        Login
      </button>
    </form>
  </section>
</div>
`;
